crates/aptos-rosetta/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! Aptos Rosetta API
+//!
+//! [Rosetta API Spec](https://www.rosetta-api.org/docs/Reference.html)
+
+use aptos_api::runtime::WebServer;
+use aptos_config::config::ApiConfig;
+use aptos_rest_client::aptos_api_types::Error;
+use aptos_types::chain_id::ChainId;
+use std::convert::Infallible;
+use tokio::task::JoinHandle;
+use warp::{
+    http::{HeaderValue, Method, StatusCode},
+    reject::{MethodNotAllowed, PayloadTooLarge, UnsupportedMediaType},
+    reply, Filter, Rejection, Reply,
+};
+
+mod account;
+pub mod client;
+pub mod common;
+pub mod error;
+pub mod types;
+
+pub const CURRENCY: &str = "APTOS";
+pub const NUM_DECIMALS: u64 = 6;
```

Comment:
Are these hardcoded just for tests?

Reply:
I'm going to come back and make these come from the actual rest API, but I'm trying to get the skeleton all in before I go ahead and start tweaking the APIs or writing custom transactions for my needs.